python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,417 @@
+            kwargs["exit_slit"]
+            if "exit_slit" in kwargs and kwargs["exit_slit"] is not None
+            else None
+        )
+
+        await self.rem.atmonochromator.cmd_updateMonochromatorSetup.set_start(
+            wavelength=self.wavelength,
+            gratingType=mono_grating,
+            fontExitSlitWidth=exit_slit,
+            fontEntranceSlitWidth=entrance_slit,
+            timeout=self.long_timeout,
+        )
+
+    async def electrometer_scan(self, duration: float) -> salobj.type_hints.BaseMsgType:
+        """Perform an electrometer scan for the specified duration.
+
+        Parameters
+        ----------
+        duration : `float`
+            Total duration of scan.
+
+        Returns
+        -------
+        RunTimeError
+
+        """
+
+        self.rem.electrometer_1.evt_largeFileObjectAvailable.flush()
+
+        try:
+            await self.electrometer.cmd_startScanDt.set_start(
+                scanDuration=duration, timeout=duration + self.long_timeout
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        # Make sure that a new lfo was created
+        lfo = await self.electrometer.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def take_fiber_spectrum(
+        self,
+        delay: float,
+        integration_time: float,
+    ) -> salobj.type_hints.BaseMsgType:
+        """Wait, then start an acquisition with the fiber spectrograph.
+
+        By default, this method will wait for `delay` seconds then start
+        an acquisition with the fiber spectrograph. Optionally the user may
+        provide a coroutine that will be awaited before the delay starts.
+
+        Parameters
+        ----------
+        delay : `float`
+            Seconds to wait before starting fiber spectrograph acquisition.
+        integration_time : `float`
+            Integration time for the fiber spectrum (seconds).
+
+        Returns
+        -------
+        RuntimeError : If exposure or return of lfo doesn't occur.
+        """
+        self.fiberspectrograph.evt_largeFileObjectAvailable.flush()
+
+        try:
+            await self.fiberspectrograph.cmd_expose.set_start(
+                integrationTime=integration_time,
+                timeout=integration_time + self.long_timeout,
+            )
+        except salobj.AckTimeoutError:
+            self.log.exception("Timed out waiting for the command ack. Continuing.")
+
+        lfo = await self.fiberspectrograph.evt_largeFileObjectAvailable.next(
+            timeout=self.long_timeout, flush=False
+        )
+        return lfo
+
+    async def perform_flat(self, exptime_dict: dict[str, float]) -> None:
+        # check dictionary
+        # Get the image number
+        # Somehow need to make sure camera is setup
+        if self.atcamera is not None:
```

Comment:
I think that, instead of the `if` statement here you can do an

```
assert self.lattis is not None
```

and just remove `if`.. This will make sure that the method will raise an exception if the camera is not set and will make `mypy` happy. 

Also, note that I am assuming you renamed `atcamera` -> `latiss`.

Reply:
Ok done. Also need to make sure that Latiss was configured with the correct filter, etc before this. I can do the checking in atcalsys, but should the actual configuration of the camera occur in atcalsys or in the script?